homeassistant/components/sensor/monzo.py

```diff
@@ -0,0 +1,157 @@
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['monzo==0.5.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ACCOUNT_ID = 'account_id'
+
+ICON_EUR = 'mdi:currency-eur'
+ICON_GBP = 'mdi:currency-gbp'
+ICON_INR = 'mdi:currency-inr'
+ICON_NGN = 'mdi:currency-ngn'
+ICON_RUB = 'mdi:currency-rub'
+ICON_TRY = 'mdi:currency-try'
+ICON_USD = 'mdi:currency-usd'
+
+DEFAULT_NAME = 'Monzo'
+
+VARIABLE_TYPES = {
+    'balance': 'Balance',
+    'currency': 'Currency',
+    'spend_today': 'Spend today',
+    'local_currency': 'Local currency',
+    'local_exchange_rate': 'Local exchange rate'
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Required(CONF_MONITORED_VARIABLES, default=[]):
+        vol.All(cv.ensure_list, [vol.In(VARIABLE_TYPES)]),
+    vol.Optional(CONF_ACCOUNT_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Monzo sensor."""
+    access_token = config.get(CONF_ACCESS_TOKEN)
+    account_id = config.get(CONF_ACCOUNT_ID)
+    name = config.get(CONF_NAME)
+
+    data = MonzoData(access_token, account_id)
+    dev = []
+    for variable in config[CONF_MONITORED_VARIABLES]:
+        dev.append(MonzoSensor(data, variable, name))
+
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class MonzoSensor(Entity):
+    """Representation of a Monzo sensor."""
+
+    def __init__(self, data, variable, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self.data = data
+        self.type = variable
+        self._name = VARIABLE_TYPES[variable]
+        self._unit_of_measurement = None
+        self._state = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        icon = ICON_USD
+
+        if not self._unit_of_measurement:
+            return icon
+        elif self._unit_of_measurement.lower() == 'eur':
+            icon = ICON_EUR
+        elif self._unit_of_measurement.lower() == 'gbp':
+            icon = ICON_GBP
+        elif self._unit_of_measurement.lower() == 'inr':
+            icon = ICON_INR
+        elif self._unit_of_measurement.lower() == 'ngn':
+            icon = ICON_NGN
+        elif self._unit_of_measurement.lower() == 'rub':
+            icon = ICON_RUB
+        elif self._unit_of_measurement.lower() == 'try':
+            icon = ICON_TRY
```

Comment:
This section could be simplified by using a directory lookup. An idea...

```python
ICON = {
 'eur': 'mdi:currency-eur',
 'gbp': 'mdi:currency-gbp',
 'inr': 'mdi:currency-inr',
 'ngn': 'mdi:currency-ngn',
 'rub': 'mdi:currency-rub',
 'try': 'mdi:currency-try',
 'usd': 'mdi:currency-usd',
}
```
then
```python
icon = ([k for k, v in ICON.items() if self._unit_of_measurement.lower() in v][0])
```
No need to touch the code if new currencies are getting available.